pkg/action/upgrade_test.go

```diff
@@ -223,7 +223,6 @@ func TestUpgradeRelease_ReuseValues(t *testing.T) {
 		expectedValues := map[string]interface{}{
 			"subchart": map[string]interface{}{
 				"enabled": false,
-				"global":  map[string]interface{}{},
```

Comment:
In this case, whatever Helm 2 does, should be what this does. Global is generally a top level field though. Just to make sure I understand, the DeepCopy fix caused this test to start failing, correct? Or is this a separate issue?

Reply:
Correct, I missed to mention that. The above test failed due to this PR's code

Reply:
I'll reproduce such an install in helm2 and see what values are stored. My guess is - it will not have global. Will check it. Will also figure out why it came up in the first place and how it got removed with this fix 😄

Reply:
In latest helm2, it doesn't store `global` field. And how it came up - not sure. My guess is, since `subchart` seems to be a key with nested value, and overriding value is nested, similar to issue #6659 , during some coalescing, the global key came up due to the absence of deep copy, so references were there. And now with this fix, it goes away

Reply:
I think we are good with this change. What do you think?

Reply:
Yep, I'm good with it!